test/e2e/daemon_set.go

```diff
@@ -61,8 +62,121 @@ var _ = Describe("Daemon set", func() {
+		label := map[string]string{daemonsetNameLabel: simpleDSName}
+
+		Logf("Creating simple daemon set %s", simpleDSName)
+		_, err := c.DaemonSets(ns).Create(&experimental.DaemonSet{
+			ObjectMeta: api.ObjectMeta{
+				Name: simpleDSName,
+			},
+			Spec: experimental.DaemonSetSpec{
+				Template: &api.PodTemplateSpec{
+					ObjectMeta: api.ObjectMeta{
+						Labels: label,
+					},
+					Spec: api.PodSpec{
+						Containers: []api.Container{
+							{
+								Name:  simpleDSName,
+								Image: image,
+								Ports: []api.ContainerPort{{ContainerPort: 9376}},
+							},
+						},
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			Logf("Check that reaper kills all daemon pods for %s", simpleDSName)
+			dsReaper, err := kubectl.ReaperFor("DaemonSet", c)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = dsReaper.Stop(ns, simpleDSName, 0, nil)
+			Expect(err).NotTo(HaveOccurred())
+			err = wait.Poll(dsRetryPeriod, dsRetryTimeout, checkRunningOnNoNodes(f, label))
+			Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to be reaped")
+		}()
+
+		By("Check that daemon pods launch on every node of the cluster.")
+		Expect(err).NotTo(HaveOccurred())
+		err = wait.Poll(dsRetryPeriod, dsRetryTimeout, checkRunningOnAllNodes(f, label))
+		Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to start")
+
+		By("Stop a daemon pod, check that the daemon pod is revived.")
+		podClient := c.Pods(ns)
+
+		podList, err := podClient.List(labels.Set(label).AsSelector(), fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(podList.Items)).To(BeNumerically(">", 0))
+		pod := podList.Items[0]
+		err = podClient.Delete(pod.Name, nil)
+		Expect(err).NotTo(HaveOccurred())
+		err = wait.Poll(dsRetryPeriod, dsRetryTimeout, checkRunningOnAllNodes(f, label))
+		Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to revive")
+
+	})
+
+	It("should run and stop complex daemon", func() {
+		ns := f.Namespace.Name
+		c := f.Client
+		complexDSName := "complex-daemon-set"
+		complexLabel := map[string]string{daemonsetNameLabel: complexDSName}
+		nodeSelector := map[string]string{daemonsetColorLabel: "blue"}
+		Logf("Creating daemon with a node selector %s", complexDSName)
+		_, err := c.DaemonSets(ns).Create(&experimental.DaemonSet{
+			ObjectMeta: api.ObjectMeta{
+				Name: complexDSName,
+			},
+			Spec: experimental.DaemonSetSpec{
+				Selector: complexLabel,
+				Template: &api.PodTemplateSpec{
+					ObjectMeta: api.ObjectMeta{
+						Labels: complexLabel,
+					},
+					Spec: api.PodSpec{
+						NodeSelector: nodeSelector,
+						Containers: []api.Container{
+							{
+								Name:  complexDSName,
+								Image: image,
+								Ports: []api.ContainerPort{{ContainerPort: 9376}},
+							},
+						},
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Initially, daemon pods should not be running on any nodes.")
```

Comment:
Can we be consistent and check this in both tests?  Why do we check it here but not in the other test?